src/pae/ieee802_1x_kay.c

```diff
@@ -688,15 +743,27 @@ ieee802_1x_kay_move_live_peer(struct ieee802_1x_mka_participant *participant,
 	ieee802_1x_kay_dump_peer(peer);
 
 	dl_list_del(&peer->list);
-	if (secy_create_receive_sc(participant->kay, rxsc)) {
+	if (secy_create_receive_sc(participant->kay, new_rxsc)) {
 		wpa_printf(MSG_ERROR, "KaY: Can't create SC, discard peer");
-		os_free(rxsc);
+		os_free(new_rxsc);
 		os_free(peer);
 		return NULL;
 	}
 	dl_list_add_tail(&participant->live_peers, &peer->list);
 
-	dl_list_add(&participant->rxsc_list, &rxsc->list);
+	/* Keep rxsc_list sorted by SCI */
+	dl_list_for_each(rxsc, &participant->rxsc_list, struct receive_sc,
+			 list) {
+		if (os_memcmp(&new_rxsc->sci, &rxsc->sci,
+			      sizeof(struct ieee802_1x_mka_sci)) > 0) {
+			dl_list_add(&rxsc->list, &new_rxsc->list);
+			found = TRUE;
```

Comment:
>TRUE [](start = 11, length = 4)

is this defined? does it pass the build?

Reply:
Current sonic-wpa-supplicant repo is not integrated with sonic-buildimage for build verification purposes and contains newer version of wpa-supplicant than debian wpa-2.9.0 creating merge conflicts.